posts/2023-08-23-liberty-olo-rh-sso.adoc

```diff
@@ -0,0 +1,394 @@
+
+Point to your image location in this section
+[source]
+----
+applicationImage: >-
+----
+and update the pullSecret needed to access the image here
+[source]
+----
+pullSecret: <secret_to_pull_image>
+----
+
+Also for the oidc: discoveryEndpoint - as shown here
+[source]
+----
+sso:
+    oidc:
+      - discoveryEndpoint: >-
+          https://keycloak-rh-sso.apps.<cluster-name>/auth/realms/sso-realm/.well-known/openid-configuration
+----
+
+point to the OpenID Endpoint Configuration that you configured while configuring the RH-SSO Operator.
+
+Here is the yaml file needed to deploy the application:
+
+[source]
+----
+apiVersion: apps.openliberty.io/v1
+kind: OpenLibertyApplication
+metadata:
+  name: guide-social-media-login
+  namespace: gsm-test
+spec:
+  sso:
+    oidc:
+      - discoveryEndpoint: >-
+          https://keycloak-rh-sso.apps.<cluster-name>/auth/realms/sso-realm/.well-known/openid-configuration
+  service:
+    port: 9443
+  applicationImage: >-
+    <image location of the app>
+  expose: true
+  manageTLS: true
+  replicas: 1
+  applicationName: guide-sm-login
+  pullPolicy: Always
+  pullSecret: <secret_to_pull_image>
+
+
+----
+== Create the OIDC Client in RH-SSO
+
+Since we have already deployed the guide-social-media-login app using the Open Liberty operator, we can now complete the registration for the openid client.
+
+Use this url - https://keycloak-rh-sso.apps.<cluster-name>/auth/admin/master/console/
+using credentials from the secret - credential-example-keycloak
+
+Click on Create to create a new client, with clientId as `gsmapp`. (The same value that you put in the secret created called `guide-social-media-login-olapp-sso`).
+
+Click on Save, which will take you to the next page which has the settings.
+On this page, ensure the default setting on 'Enabled' which needs to be 'ON' to ensure the client is enabled for login, and 'Access Type' as 'public' doesn't require a secret for login.
+
+Enter the URL for Valid Redirect URIs. In the scenario with 'oidcLogin', the URL will be https://<app-name>-<namespace>.apps.<cluster-name>/ibm/api/social-login/redirect/oidc
+
+For my test, I put in this value
+
+https://guide-social-media-login-gsm-test.apps.<cluster-name>/ibm/api/social-login/redirect/oidc
+and clicked on Save.
+
+== Running the application, and logging in using OIDC
+
+
+Since all the configuration is complete, you are ready to run the application now.
+Get the route of the application from the gsm-test project.
+It should be in this format:
+
+https://guide-social-media-login-gsm-test.apps.<cluster-name>/guide-social-login/hello.html
+
+You should see the application as shown below.
+
+[.img_border_light]
+image::img/blog/rh_social_media_login.png[Social Media Login,width=50%,align="center"]
+
+
+Since I have already created the RH-SSO client for this application, when I click on the "Log In" button for this app, it will redirect me to the RH-SSO client, as shown below.
```

Comment:
Use of I again, doesn't work with tenses used within rest of blog. Also isn't engaging for readers.

Reply:
This seems to be the crux of the whole article (i.e. what we've been trying to achieve this whole time), but it feels like a throwaway side comment with the short length. I think this section should have more prose and more impact.